crategen/models/tes_models.py

```diff
@@ -0,0 +1,283 @@
+        """- If content is set url should be ignored.
+
+        - If content is not set then url should be present.
+        """
+        content_is_set = (
+            values.get("content") and len(values.get("content").strip()) > 0
+        )
+        url_is_set = values.get("url") and len(values.get("url").strip()) > 0
+
+        if content_is_set:
+            values["url"] = None
+        elif not content_is_set and not url_is_set:
+            raise ValueError(
+                "The 'url' attribute is required when the 'content' attribute is empty"
+            )
+        return values
+
+    @validator("path")
+    def validate_path(cls, value):
+        """Validate that the path is an absolute path."""
+        if not os.path.isabs(value):
+            raise ValueError("The 'path' attribute must contain an absolute path.")
+        return value
+
+
+class TESOutput(BaseModel):
+    """Output files. Outputs will be uploaded from the executor container to long-term storage.
+
+    **Attributes:**
+
+    - **name** (`Optional[str]`): User-provided name of output file
+    - **description** (`Optional[str]`): Optional users provided description field, can be used for documentation.
+    - **url** (`AnyUrl`): URL for the file to be copied by the TES server after the task is complete
+    - **path** (`str`): Path of the file inside the container. Must be an absolute path.
+    - **type** (`TESFileType`): The type of output (e.g., FILE, DIRECTORY).
+
+    Reference: https://ga4gh.github.io/task-execution-schemas/docs/#operation/GetTask
+    """
+
+    name: Optional[str] = None
+    description: Optional[str] = None
+    url: AnyUrl
+    path: str
+    type: TESFileType = TESFileType.FILE
+
+    @validator("path")
+    def validate_path(cls, value):
+        """Validate that the path is an absolute path."""
+        if not os.path.isabs(value):
+            raise ValueError("The 'path' attribute must contain an absolute path.")
+        return value
+
+
+class TESTaskLog(BaseModel):
+    """Task logging information. Normally, this will contain only one entry, but in the case where a task fails and is retried, an entry will be appended to this list.
+
+    **Attributes:**
+
+    - **logs** (`list[TESExecutorLog]`): Logs for each executor.
+    - **metadata** (`Optional[dict[str, str]]`): Arbitrary logging metadata included by the implementation.
+    - **start_time** (`Optional[datetime]`): When the task started, in RFC 3339 format.
+    - **end_time** (`Optional[datetime]`): When the task ended, in RFC 3339 format.
+    - **outputs** (`list[TESOutputFileLog]`): Information about all output files. Directory outputs are flattened into separate items.
+    - **system_logs** (`Optional[list[str]]`): System logs are any logs the system decides are relevant, which are not tied directly to an Executor process. Content is implementation specific: format, size, etc.
+    - **status** (`Optional[str]`): The status of the task.
```

Comment:
Indeed, `state` (not `status`) is a property of `tesTask` (`TESData` in your language), not `TESTaskLog` - so please remove from the docstring.

BTW: Also note that, again, `state` is, of course, implicitly required when requesting a task resource from the service, but it is not allowed to be set by the user. Ideally, the spec would use `readOnly` and `writeOnly` properties for these, but they were only added in OpenAPI 3, and the specs were migrated from OpenAPI 2, without adding any of the OpenAPI 3 features.

Not quite sure how to deal with such issues, given the implicit/ambiguous nature of such props. For now I guess it's best to just leave it optional.

Reply:
I noticed the mention of `status` in the docstring of the `TESTaskLog` class, which is not present in the class attributes. Based on your feedback and the TES v1.1 schema, I will remove `status` from the docstring to avoid confusion.

Regarding the `state` field, I will ensure it is correctly implemented in `TESData` and marked as optional since it is set by the service and not by the user.